### PR TITLE
add vertex validation if vertex is available

### PR DESCRIPTION
**Pull request by hkernbach**
### Scope & Purpose

This PR adds further validation to Graphs when editing edge definitions.
**There is also an EE PR.** https://github.com/arangodb/enterprise/pull/500

- [x] Bug-Fix for *devel-branch*
Yes, backports are required for all supported versions.

#### Related Information

- [x] There is an internal planning ticket: https://arangodb.atlassian.net/browse/QA-23

### Testing & Verification

This PR adds tests that were used to verify all changes:

- [x] Added new **integration tests** (in shell_server (EE))

### CODE REVIEW

File: arangod/Graph/GraphManager.h [hunk 1]
```diff
@@ -93,12 +93,13 @@ class GraphManager {
...
-      std::map<std::string, EdgeDefinition> const& edgeDefinitions,
+      Graph const& graph, std::map<std::string, EdgeDefinition> const& edgeDefinitions,
       bool waitForSync, VPackSlice options);
 
-  OperationResult findOrCreateCollectionsByEdgeDefinition(EdgeDefinition const& edgeDefinition,
+  OperationResult findOrCreateCollectionsByEdgeDefinition(Graph const& graph,
+                                                          EdgeDefinition const& edgeDefinition,
                                                           bool waitForSync,
-                                                          VPackSlice options);
+                                                          VPackSlice const options);
```
Comment (goedderz) [inline arangod/Graph/GraphManager.h:102]:
```suggestion
                                                          VPackSlice options);
```
Adding `const` at the function declaration does make no difference, as the parameter is passed by value and thus the function cannot modify the original value anyway. It's good to leave it at the definition, though, where it prohibits to modify the local variable.

**Comment by hkernbach** [comment]
http://jenkins01:8080/view/PR/job/arangodb-matrix-pr/10934/

**Comment by hkernbach** [comment]
Jenkins green.

File: CHANGELOG [hunk 2]
```diff
@@ -1,6 +1,8 @@
 devel
 -----
 
+* Added vertex validation in case of a SmartGraph edge definition update.
```
Comment (goedderz) [inline CHANGELOG:4]:
```suggestion
* Added vertex collection validation in case of a SmartGraph edge definition update. 
```

**Pull request merged**